src/timeline/components/PrevSessionContext.jsx

```diff
@@ -1,19 +1,37 @@
 const React = require("react");
-const { useContext, createContext, useState } = React;
 
-// Create the context
+const { useContext, createContext, useState, useEffect } = React;
+const { AllStudySessionProcessing } = require("../../api_recievers/influxqueries");
+
+
 const PrevSessionContext = createContext();
 
-// Create a provider component
-const PrevSessionProvider = ({children}) => {
-  const prevSessions = [
-    { name: "MM/DD/YY HH:MM to HH:MM", sessionId: 0, duration: 10000 },
-    { name: "MM/DD/YY HH:MM to HH:MM", sessionId: 1, duration: 10000 }
-  ];
-  
-  //REPLACE THIS ARRAY with the data for each study session as queries from the database
-  
-  const [prevSessionIds, setPrevSessionIds] = useState(prevSessions);
+const PrevSessionProvider = ({ children }) => {
+  const [prevSessionIds, setPrevSessionIds] = useState([]);
+
+  useEffect(() => {
+    
+    AllStudySessionProcessing()
```

Comment:
This is super nitpicky but I'd prefer if you used await here

Reply:
Oh wait I forgot useEffect gets angry if you try to make the callback async nvm this is valid